Slim/Schema/Album.pm

```diff
@@ -259,6 +259,31 @@ sub artistsForRoles {
 		->search_related('contributor')->distinct->all;
 }
 
+sub artistPerformsOnWork {
+	my ($self, $work, $grouping, $artist) = @_;
+
+		my $sth = Slim::Schema->dbh->prepare_cached(
+			qq{
+				SELECT count(*)
+				from albums
+				JOIN tracks ON albums.id = tracks.album
+				JOIN contributor_track ON tracks.id = contributor_track.track
+				WHERE tracks.work = :work
+				AND albums.id = :album
+				AND contributor_track.contributor = :artist
+				AND ( (:grouping IS NULL AND tracks.grouping IS NULL) OR tracks.grouping = :grouping )
+			}
+		);
+		$sth->bind_param(":work", $work);
+		$sth->bind_param(":album", $self->id);
+		$sth->bind_param(":artist", $artist);
+		$sth->bind_param(":grouping", $grouping);
+		$sth->execute();
+		my ($count) = $sth->fetchrow_array;
+		$sth->finish;
+		return $count
```

Comment:
This would only ever be called if we dealt with a single work, right? Otherwise the additional requests could become somewhat of a burden.

Please shift left the block by one tab. Oh, and while I'm picky: please uppercase `FROM` in the select 😁.

Reply:
Yes, only a single work, from the AlbumInfo filter.